actors.md

```diff
@@ -324,13 +324,35 @@ func SlashConsensusFault(block1, block2 BlockHeader) {
 		Fatal("blocks do not prove a slashable offense")
 	}
 
+	if !self.Miners.Has(msg.From) {
+		Fatal("slash collateral must only be called by a miner actor")
+	}
+
 	miner := AuthorOf(block1)
 
-	// TODO: Some of the slashed collateral should be paid to the slasher
+	// Check if miner has been slashed
+	if !self.Miners.Has(miner) {
+		Fatal("miner has been slashed")
+	}
 
 	// Burn all of the miners collateral
-	miner.BurnCollateral()
-
+	// const slashedCollateral = self.GetCollateral(miner)
+	const slashedCollateral = miner.Balance
```

Comment:
I thought we weren't slashing the entire balance on a fault? Can we leave a placeholder here for the amount slashed, e.g. something like:

`miner.Balance -= slashingAmountForFault(fault)`

Reply:
@decentralion I think we might be slashing the entire amount here if it is consensus fault. Otherwise, we have to decide how much to slash. cc @whyrusleeping

Reply:
A miner's balance probably be greater than their collateral requirements, but could be less. We should slash the min of those two. IMO we should not burn more than the miner's collateral requirement just because they happen to have that on balance.

Reply:
@anorth resolved in latest commit